interview-questions.md

```diff
@@ -8,52 +8,86 @@ Then, research the question to expand on your answer. Even if you feel you have
+  
+  Another type are Columnar databases, also referred to as column data store. They store data in columns rather than rows and are often used in data warehouses. Some examples would be Google BigQuery, and Azure SQL Data Warehouse. 
+  
+  There are also wide column databases, object oriented databases, key-value databases, hierarchial databases, Document databases, graph databases, and time series databases.
 
 5. What are primary keys? Why are they important?
 
   Your answer:
+  Primary keys are how we identifty things in Active Record. It is created by SQL when making tables of data,
 
   Researched answer:
 
+Primary key identifies each record in a table. A table can only have one primary key  and it can be made of single or multiple columns. I primary that consists real world observables is called a natural key. There are also surrogate keys which function as a key and are not used for id purposes outside the database. Primary keys are identified in SQL by the PRIMARY KEY constraint. 
+
+
+
 
 
 ## Looking Ahead: Terms for Next Week
 Research and define the following terms to the best of your ability.
 
 1. RESTful routes:
 
+REST stands for Representatuinal State Transfer. RESTful routing is a set of standards used in many different languages to create efficient and reuable routes. It mixes HTTP methods and CRUD actions together so that it's easier for other developers to understand and navigate and app and it's results. 
+
+
 2. JSON:
 
+JSON, or JavaScript Object Notation,  is a data-interchanging format that is easy for humans to read and write. It's also easy for machines to generatge and is based on a subset of the JS language. 
+JSON is built on two structures. A collection of name/value pairs and an ordered list of values. 
+
 3. ERB:
+  ERB stands for Epic Rap Battles of History..Just kidding thats the first thing that popped up when Googling "ERB".
+
+  Anyways, ERB is a feature of Ruby that enables you to generate any kind of text, in any quantity, from "ERB templates". ERB templates combine Ruby code with plain text. The Ruby code is for variable substitution and flow control. 
 
 4. Params:
+Params, a keyword, is a way that a developer can specify a method parameter that takes a variable number of arguments. It also allows you to call a function with no arguments. You can use params when the number of arguments can range from 0 to infinity.
 
 5. API:
+API, short for application programming interface, is a set of definitions and protocols for building and integrating app software. An API is a platform that allows different pieces of software or applications to communicate with one another. An example would be PayPal or weather snippets, which are found on every platform. To specify, the weather snippest are the dedicated boxes that are found when one Googles `weather for ${currentCity}` or the weather app on Apple products.
```

Comment:
Great job on the looking ahead section. It is really helpful to spend some time with these concepts prior to lecture. Also, this made me laugh: `ERB stands for Epic Rap Battles of History..Just kidding thats the first thing that popped up when Googling "ERB".` :)